### PR TITLE
update transformers version to 0.6

### PR DESCRIPTION
**Pull request by epoberezkin**
Can we do something like this?

Or even just adding 0.6 in the range would be helpful, as otherwise it doesn't compile with GHC 9.6.2

Switched to fork for now.

Thank you!

### CODE REVIEW

File: network-transport.cabal [hunk 1]
```diff
@@ -68,7 +68,7 @@ Library
                    binary >= 0.5 && < 0.9,
                    bytestring >= 0.9 && < 0.12,
                    hashable >= 1.2.0.5 && < 1.5,
-                   transformers >= 0.2 && < 0.6,
+                   transformers >= 0.6 && < 0.7,
```
Comment (facundominguez) [inline network-transport.cabal:71]: Thanks @epoberezkin! Is the update to the lower bound necessary?
```suggestion
                   transformers >= 0.2 && < 0.7,
```

**Comment by Bodigrim** [comment]
As a Hackage trustee I made a relaxing revision: https://hackage.haskell.org/package/network-transport-0.5.6/revisions/
(There is no need to prohibit `transformers < 0.6` indeed)

**Comment by facundominguez** [comment]
Thanks @Bodigrim, I'm merging and restoring the old lowerbound afterwards.

**Comment by facundominguez** [comment]
Restored the old lowerbound in fbcb151.

**Pull request merged**